### ISSUE TITLE
Segment fault when train with binary_cross_entropy

### ISSUE THREAD

**Issue by MintYiqingchen** [Microsoft/CNTK#3027]
I write a loss function as following:
` 
cls_loss = C.binary_cross_entropy(mod_cls_logits,slc, name='classify')                                               
# span loss [#][1] + cls loss [#][1]
new_loss = all_spans_loss(start_logits, ab, end_logits, ae)*slc + C.constant(10)*cls_loss
`
when I train my model using new loss, it can work smoothly, but when I use only cls_loss as my loss function, it will cause segment fault.But cls_loss.eval(data_map) works fine.

The fault information:

> [sdcs-d1-03:22567] *** Process received signal ***
> [sdcs-d1-03:22567] Signal: Segmentation fault (11)
> [sdcs-d1-03:22567] Signal code: Address not mapped (1)
> [sdcs-d1-03:22567] Failing at address: 0x8
> [sdcs-d1-03:22567] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x10330)[0x7f6d44eeb330]
> [sdcs-d1-03:22567] [ 1] /home/keze/anaconda2/envs/cntk-py34/lib/python3.4/site-packages/cntk/libs/libCntk.Math-2.3.1.so(_ZNK9Microsoft3MSR4CNTK6MatrixIfE10GetNumColsEv+0x0)[0x7f6cfe6ca530]
> [sdcs-d1-03:22567] [ 2] /home/keze/anaconda2/envs/cntk-py34/lib/python3.4/site-packages/cntk/libs/libCntk.Core-2.3.1.so(_ZN4CNTK17CompositeFunction23GetNodeOutputOrGradientENS_8VariableERSt10shared_ptrINS_5ValueEERS2_IN9Microsoft3MSR4CNTK19ComputationNodeBaseEEb+0x3b8)[0x7f6d038ea118]
> [sdcs-d1-03:22567] [ 3] /home/keze/anaconda2/envs/cntk-py34/lib/python3.4/site-packages/cntk/libs/libCntk.Core-2.3.1.so(_ZN4CNTK17CompositeFunction19GetNetworkGradientsERSt13unordered_mapINS_8VariableESt10shared_ptrINS_5ValueEESt4hashIS2_ESt8equal_toIS2_ESaISt4pairIKS2_S5_EEE+0x127)[0x7f6d038ea7f7]
> [sdcs-d1-03:22567] [ 4] /home/keze/anaconda2/envs/cntk-py34/lib/python3.4/site-packages/cntk/libs/libCntk.Core-2.3.1.so(_ZN4CNTK17CompositeFunction8BackwardERKSt10shared_ptrINS_13BackPropStateEERKSt13unordered_mapINS_8VariableES1_INS_5ValueEESt4hashIS7_ESt8equal_toIS7_ESaISt4pairIKS7_S9_EEERSI_+0x2cf)[0x7f6d038ead4f]
> [sdcs-d1-03:22567] [ 5] /home/keze/anaconda2/envs/cntk-py34/lib/python3.4/site-packages/cntk/libs/libCntk.Core-2.3.1.so(_ZN4CNTK7Trainer22ExecuteForwardBackwardERKSt13unordered_mapINS_8VariableESt10shared_ptrINS_5ValueEESt4hashIS2_ESt8equal_toIS2_ESaISt4pairIKS2_S5_EEERSE_RKNS_16DeviceDescriptorESH_+0x5d7)[0x7f6d0396e027]
> [sdcs-d1-03:22567] [ 6] /home/keze/anaconda2/envs/cntk-py34/lib/python3.4/site-packages/cntk/libs/libCntk.Core-2.3.1.so(_ZN4CNTK7Trainer19TrainLocalMinibatchERKSt13unordered_mapINS_8VariableESt10shared_ptrINS_5ValueEESt4hashIS2_ESt8equal_toIS2_ESaISt4pairIKS2_S5_EEERSE_bRKNS_16DeviceDescriptorE+0xd3)[0x7f6d0396e7d3]
> [sdcs-d1-03:22567] [ 7] /home/keze/anaconda2/envs/cntk-py34/lib/python3.4/site-packages/cntk/libs/libCntk.Core-2.3.1.so(_ZN4CNTK7Trainer14TrainMinibatchERKSt13unordered_mapINS_8VariableENS_13MinibatchDataESt4hashIS2_ESt8equal_toIS2_ESaISt4pairIKS2_S3_EEERS1_IS2_St10shared_ptrINS_5ValueEES5_S7_SaIS8_IS9_SH_EEERKNS_16DeviceDescriptorE+0x8b)[0x7f6d0396f25b]
> [sdcs-d1-03:22567] [ 8] /home/keze/anaconda2/envs/cntk-py34/lib/python3.4/site-packages/cntk/libs/libCntk.Core-2.3.1.so(_ZN4CNTK7Trainer14TrainMinibatchERKSt13unordered_mapINS_8VariableENS_13MinibatchDataESt4hashIS2_ESt8equal_toIS2_ESaISt4pairIKS2_S3_EEERKNS_16DeviceDescriptorE+0xa0)[0x7f6d0396f400]
> [sdcs-d1-03:22567] [ 9] /home/keze/anaconda2/envs/cntk-py34/lib/python3.4/site-packages/_cntk_py.cpython-34m.so(+0x265f2e)[0x7f6d0445cf2e]
> [sdcs-d1-03:22567] [10] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyEval_EvalFrameEx+0x5ffe)[0x7f6d45229d8e]
> [sdcs-d1-03:22567] [11] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyEval_EvalCodeEx+0x8b1)[0x7f6d4522c371]
> [sdcs-d1-03:22567] [12] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyEval_EvalFrameEx+0x6f9a)[0x7f6d4522ad2a]
> [sdcs-d1-03:22567] [13] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyEval_EvalCodeEx+0x8b1)[0x7f6d4522c371]
> [sdcs-d1-03:22567] [14] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyEval_EvalFrameEx+0x6f9a)[0x7f6d4522ad2a]
> [sdcs-d1-03:22567] [15] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyEval_EvalCodeEx+0x8b1)[0x7f6d4522c371]
> [sdcs-d1-03:22567] [16] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyEval_EvalFrameEx+0x6f9a)[0x7f6d4522ad2a]
> [sdcs-d1-03:22567] [17] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyEval_EvalCodeEx+0x8b1)[0x7f6d4522c371]
> [sdcs-d1-03:22567] [18] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyEval_EvalCode+0x3b)[0x7f6d4522c43b]
> [sdcs-d1-03:22567] [19] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyRun_FileExFlags+0x130)[0x7f6d4524cdb0]
> [sdcs-d1-03:22567] [20] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(PyRun_SimpleFileExFlags+0x104)[0x7f6d4524fe44]
> [sdcs-d1-03:22567] [21] /home/keze/anaconda2/envs/cntk-py34/bin/../lib/libpython3.4m.so.1.0(Py_Main+0xba6)[0x7f6d4526a276]
> [sdcs-d1-03:22567] [22] python(main+0x15d)[0x400acd]
> [sdcs-d1-03:22567] [23] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf5)[0x7f6d44426f45]
> [sdcs-d1-03:22567] [24] python[0x4008a9]
> [sdcs-d1-03:22567] *** End of error message ***
> Segmentation fault (core dumped)

**Comment by KeDengMS** [comment]
Thanks for the report. Can you share a repro script?
